Build error on bare system - building from source

## Describe the bug?

Trying to build the repo from source throws errors about SSE support. Here is the error in its entirety:

## Steps to reproduce?
1. git clone https://github.com/capsule-rs/capsule.git
2. cargo build

## Expected behavior?

Successful build

### Capsule version?

Git

### OS?

Ubuntu 18.10

### Docker / VM / Bare?

Bare

### Stack trace or error log output

`  running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-march=corei7" "-o" "/users/deep83/rust/capsule/target/debug/build/capsule-ffi-9f389138f8a5f94a/out/src/shim.o" "-c" "src/shim.c"
  cargo:warning=In file included from /usr/lib/gcc/x86_64-linux-gnu/7/include/immintrin.h:41:0,
  cargo:warning=                 from /usr/lib/gcc/x86_64-linux-gnu/7/include/x86intrin.h:48,
  cargo:warning=                 from /usr/local/include/rte_vect.h:28,
  cargo:warning=                 from /usr/local/include/rte_memcpy.h:17,
  cargo:warning=                 from /usr/local/include/rte_ether.h:21,
  cargo:warning=                 from /usr/local/include/rte_ethdev.h:159,
  cargo:warning=                 from src/shim.c:20:
  cargo:warning=/usr/local/include/rte_memcpy.h: In function ‘rte_mov32’:
  cargo:warning=/usr/lib/gcc/x86_64-linux-gnu/7/include/avxintrin.h:926:1: error: inlining failed in call to always_inline ‘_mm256_storeu_si256’: target specific option mismatch
  cargo:warning= _mm256_storeu_si256 (__m256i_u *__P, __m256i __A)
  cargo:warning= ^~~~~~~~~~~~~~~~~~~
  cargo:warning=In file included from /usr/local/include/rte_ether.h:21:0,
  cargo:warning=                 from /usr/local/include/rte_ethdev.h:159,
  cargo:warning=                 from src/shim.c:20:
  cargo:warning=/usr/local/include/rte_memcpy.h:321:2: note: called from here
  cargo:warning=  _mm256_storeu_si256((__m256i *)dst, ymm0);
  cargo:warning=  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=In file included from /usr/lib/gcc/x86_64-linux-gnu/7/include/immintrin.h:41:0,
  cargo:warning=                 from /usr/lib/gcc/x86_64-linux-gnu/7/include/x86intrin.h:48,
  cargo:warning=                 from /usr/local/include/rte_vect.h:28,
  cargo:warning=                 from /usr/local/include/rte_memcpy.h:17,
  cargo:warning=                 from /usr/local/include/rte_ether.h:21,
  cargo:warning=                 from /usr/local/include/rte_ethdev.h:159,
  cargo:warning=                 from src/shim.c:20:
  cargo:warning=/usr/lib/gcc/x86_64-linux-gnu/7/include/avxintrin.h:920:1: error: inlining failed in call to always_inline ‘_mm256_loadu_si256’: target specific option mismatch
  cargo:warning= _mm256_loadu_si256 (__m256i_u const *__P)
  cargo:warning= ^~~~~~~~~~~~~~~~~~
  cargo:warning=In file included from /usr/local/include/rte_ether.h:21:0,
  cargo:warning=                 from /usr/local/include/rte_ethdev.h:159,
  cargo:warning=                 from src/shim.c:20:
  cargo:warning=/usr/local/include/rte_memcpy.h:320:7: note: called from here
  cargo:warning=  ymm0 = _mm256_loadu_si256((const __m256i *)src);
  cargo:warning=  ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  exit code: 1

  --- stderr


  error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-march=corei7" "-o" "/users/deep83/rust/capsule/target/debug/build/capsule-ffi-9f389138f8a5f94a/out/src/shim.o" "-c" "src/shim.c" with args "cc" did not execute successfully (status code exit code: 1).`

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -292,6 +292,10 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
         continue-on-error: true
 
+  ## TODO 01/23/21: Return to this, as it's getting an Exit 1,
+  ##      but I can't get leak errors locally on our VM (exit 0) or my local
+  ##      system 76 (bare). So, only shows up on GH (no problem in past runs).
+  ##      I've tested on recent rust nightly and 19.11.6 (and 19.11.1)
   sanitize:
     runs-on: ubuntu-18.04
 
@@ -314,7 +318,9 @@ jobs:
             --env LSAN_OPTIONS=suppressions=suppressions.txt \
             getcapsule/sandbox \
             /bin/bash -c "rustup install nightly && rustup default nightly \
-            && cargo run --target x86_64-unknown-linux-gnu -- -f ping4d.toml"
+            && rustup component add rust-src \
+            && cargo run --target x86_64-unknown-linux-gnu -Zbuild-std -- -f ping4d.toml"
+        continue-on-error: true
 
       - name: slack-it
         uses: homoluctus/slatify@v2.1.2

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -111,6 +111,7 @@ jobs:
         with:
           name: criterion-report
           path: ./bench/target/criterion
+        continue-on-error: true
 
       - name: slack-it
         uses: homoluctus/slatify@v2.1.2

--- a/README.md
+++ b/README.md
@@ -23,35 +23,35 @@
 
 # Capsule
 
-A framework for network function development. Written in Rust, inspired by [NetBricks](https://www.usenix.org/system/files/conference/osdi16/osdi16-panda.pdf) and built on Intel's [Data Plane Development Kit](https://www.dpdk.org/). 
+A framework for network function development. Written in Rust, inspired by [NetBricks](https://www.usenix.org/system/files/conference/osdi16/osdi16-panda.pdf) and built on Intel's [Data Plane Development Kit](https://www.dpdk.org/).
 
 ## Table of Contents
 
-* [Overview](#overview)
-* [Quick Start](#quick-start)
-* [Contributing](#contributing)
-* [Code of Conduct](#code-of-conduct)
-* [Contact](#contact)
-* [Maintainers](#maintainers)
-* [License](#license)
+- [Overview](#overview)
+- [Quick Start](#quick-start)
+- [Contributing](#contributing)
+- [Code of Conduct](#code-of-conduct)
+- [Contact](#contact)
+- [Maintainers](#maintainers)
+- [License](#license)
 
 ## Overview
 
 The goal of `Capsule` is to offer an ergonomic framework for network function development that traditionally has high barriers of entry for developers. We've created a tool to efficiently manipulate network packets while being type-safe, memory-safe, and thread-safe. Building on `DPDK` and `Rust`, `Capsule` offers
 
-* a fast packet processor that uses minimum number of CPU cycles.
-* a rich packet type system that guarantees memory-safety and thread-safety.
-* a declarative programming model that emphasizes simplicity.
-* an extensible and testable framework that is easy to develop and maintain.
+- a fast packet processor that uses minimum number of CPU cycles.
+- a rich packet type system that guarantees memory-safety and thread-safety.
+- a declarative programming model that emphasizes simplicity.
+- an extensible and testable framework that is easy to develop and maintain.
 
 ## Quick Start
 
 The easiest way to start developing `Capsule` applications is to use the `Vagrant` [virtual machine](https://github.com/capsule-rs/sandbox/blob/master/Vagrantfile) and the `Docker` [sandbox](https://hub.docker.com/repository/docker/getcapsule/sandbox) provided by our team. The sandbox is preconfigured with all the necessary tools and libraries for `Capsule` development, including:
 
-* [`DPDK` 19.11](https://doc.dpdk.org/guides-19.11/rel_notes/release_19_11.html)
-* [`Clang`](https://clang.llvm.org/) and [`LLVM`](https://www.llvm.org/)
-* [`Rust 1.45`](https://blog.rust-lang.org/2020/07/16/Rust-1.45.0.html)
-* [`rr`](https://rr-project.org/) 5.3
+- [`DPDK` 19.11](https://doc.dpdk.org/guides-19.11/rel_notes/release_19_11.html)
+- [`Clang`](https://clang.llvm.org/) and [`LLVM`](https://www.llvm.org/)
+- [`Rust 1.49`](https://blog.rust-lang.org/2020/12/31/Rust-1.49.0.html)
+- [`rr`](https://rr-project.org/) 5.3
 
 First install [`Vagrant`](https://www.vagrantup.com/) and [`VirtualBox`](https://www.virtualbox.org/) on your system. Also install the following `Vagrant` plugins,
 
@@ -79,7 +79,7 @@ vagrant$ docker run -it --rm \
     --security-opt seccomp=unconfined \
     -v /lib/modules:/lib/modules \
     -v /dev/hugepages:/dev/hugepages \
-    getcapsule/sandbox:19.11.1-1.45 /bin/bash
+    getcapsule/sandbox:19.11.6-1.49 /bin/bash
 ```
 
 Remember to also mount the working directory of your project as a volume for the sandbox. Then you can use `Cargo` commands inside the container as normal.
@@ -109,14 +109,14 @@ You can contact us either through [`Discord`](https://discord.gg/sVN47RU) or [em
 
 The current maintainers with roles to merge PRs are:
 
-* [Daniel Jin](https://github.com/drunkirishcoder)
-* [Zeeshan Lakhani](https://github.com/zeeshanlakhani)
-* [Andrew Wang](https://github.com/awangc)
-* [Peter Cline](https://github.com/clinedome)
+- [Daniel Jin](https://github.com/drunkirishcoder)
+- [Zeeshan Lakhani](https://github.com/zeeshanlakhani)
+- [Andrew Wang](https://github.com/awangc)
+- [Peter Cline](https://github.com/clinedome)
 
 ## Read More About Capsule
 
-* [Modernize network function development with this Rust-based framework](https://opensource.com/article/20/8/capsule-networking)
+- [Modernize network function development with this Rust-based framework](https://opensource.com/article/20/8/capsule-networking)
 
 ## License
 

--- a/core/src/lib.rs
+++ b/core/src/lib.rs
@@ -49,7 +49,7 @@
 //!
 //! * [DPDK 19.11]
 //! * [Clang] and [LLVM]
-//! * [Rust 1.45]
+//! * [Rust 1.49]
 //! * [rr] 5.3
 //!
 //! For more information on getting started, please check out Capsule's
@@ -105,7 +105,7 @@
 //! [DPDK 19.11]: https://doc.dpdk.org/guides-19.11/rel_notes/release_19_11.html
 //! [Clang]: https://clang.llvm.org/
 //! [LLVM]: https://www.llvm.org/
-//! [Rust 1.45]: https://blog.rust-lang.org/2020/07/16/Rust-1.45.0.html
+//! [Rust 1.49]: https://blog.rust-lang.org/2020/12/31/Rust-1.49.0.html
 //! [rr]: https://rr-project.org/
 //! [README]: https://github.com/capsule-rs/capsule/blob/master/README.md
 //! [sandbox repo]: https://github.com/capsule-rs/sandbox

--- a/core/src/pcap.rs
+++ b/core/src/pcap.rs
@@ -84,6 +84,10 @@ impl Pcap {
     /// Append to already-existing file for dumping packets into from a given
     /// file path.
     fn append(path: &str) -> Fallible<Pcap> {
+        if !std::path::Path::new(path).exists() {
+            return Err(PcapError::new("Pcap filename path does not exist.").into());
+        }
+
         unsafe {
             let handle = ffi::pcap_open_dead(DLT_EN10MB, PCAP_SNAPSHOT_LEN)
                 .to_result(|_| PcapError::new("Cannot create packet capture handle."))?;
@@ -93,7 +97,6 @@ impl Pcap {
                     ffi::pcap_close(handle.as_ptr());
                     err
                 })?;
-
             Ok(Pcap {
                 path: path.to_string(),
                 handle,

--- a/core/src/runtime/mod.rs
+++ b/core/src/runtime/mod.rs
@@ -296,7 +296,7 @@ impl Runtime {
         let kni_rx = self
             .get_port_mut(port)?
             .kni()
-            .ok_or_else(|| KniError::Disabled)?
+            .ok_or(KniError::Disabled)?
             .take_rx()?;
 
         // selects a core to run a rx pipeline for this port. the selection is

--- a/ffi/build.rs
+++ b/ffi/build.rs
@@ -171,6 +171,7 @@ const RTE_DEPS_LIBS: &[&str] = &["numa", "pcap"];
 fn bind(path: &Path) {
     cc::Build::new()
         .file("src/shim.c")
+        .flag("-mavx")
         .flag("-march=corei7")
         .compile("rte_shim");
 
@@ -190,6 +191,7 @@ fn bind(path: &Path) {
         .derive_partialeq(true)
         .default_enum_style(bindgen::EnumVariation::ModuleConsts)
         .clang_arg("-finline-functions")
+        .clang_arg("-mavx")
         .clang_arg("-march=corei7")
         .rustfmt_bindings(true)
         .generate()

--- a/ffi/build.rs
+++ b/ffi/build.rs
@@ -171,8 +171,7 @@ const RTE_DEPS_LIBS: &[&str] = &["numa", "pcap"];
 fn bind(path: &Path) {
     cc::Build::new()
         .file("src/shim.c")
-        .flag("-mavx")
-        .flag("-march=corei7")
+        .flag("-march=corei7-avx")
         .compile("rte_shim");
 
     bindgen::Builder::default()
@@ -191,8 +190,7 @@ fn bind(path: &Path) {
         .derive_partialeq(true)
         .default_enum_style(bindgen::EnumVariation::ModuleConsts)
         .clang_arg("-finline-functions")
-        .clang_arg("-mavx")
-        .clang_arg("-march=corei7")
+        .clang_arg("-march=corei7-avx")
         .rustfmt_bindings(true)
         .generate()
         .expect("Unable to generate bindings")

